src/lib/comms/sol-http-common.c

```diff
@@ -843,15 +844,27 @@ sol_http_split_query(const char *query, struct sol_http_params *params)
         }
 
         if (!sol_http_param_add_copy(params,
-            CREATE_QUERY_PARAM(key, value))) {
+            CREATE_PARAM(key, value))) {
             SOL_ERR("Could not add the HTTP param %.*s:%.*s",
                 SOL_STR_SLICE_PRINT(key), SOL_STR_SLICE_PRINT(value));
             goto exit;
         }
     }
-#undef CREATE_QUERY_PARAM
+#undef CREATE_PARAM
 
 exit:
     sol_vector_clear(&tokens);
     return 0;
 }
+
+SOL_API int
+sol_http_split_query(const char *query, struct sol_http_params *params)
+{
+    return sol_http_split_str_key_value(query, SOL_HTTP_PARAM_QUERY_PARAM, params);
+}
```

Comment:
I guessed it was public already.

As for the function, it can calculate the delimiters based on the type (`SOL_HTTP_PARAM_POST_FIELD`), no? For Headers it could use `:`.